content/300-guides/050-database/870-using-prisma-with-mongodb.mdx

```diff
@@ -0,0 +1,351 @@
+
+**Scenario**: you need to add a new field to your schema to accommodate a billing address that is separate from the currently stored shipping address. You currently have the following schema in your `schema.prisma` file:
+
+```prisma file=prisma/schema.prisma
+model Product {
+  id     String  @id @default(auto()) @map("_id") @db.ObjectId
+  name   String  @unique
+}
+
+model Order {
+  id              String   @id @default(auto()) @map("_id") @db.ObjectId
+  product         Product  @relation(fields: [productId], references: [id])
+  shippingAddress Address
+  productId       String   @db.ObjectId
+}
+
+type Address {
+  street String @id @default(auto()) @map("_id") @db.ObjectId
+  city   String
+  zip    String
+}
+```
+
+There are a number of strategies you could use for migrating this schema:
+
+- **"On-demand" updates**: with this strategy, you and your team have agreed that updates can be made to the schema as needed. However, in order to avoid migration failures due to inconsistencies between the data and schema, there is agreement in the team that any new fields added are explicitly defined as optional.
+
+  In our scenario above, you can go ahead and add the new `billingAddress` field to the `User` model in your Prisma schema, and define it as optional:
+
+  ```prisma file=prisma/schema.prisma highlight=10;add
+  model Product {
+    id     String  @id @default(auto()) @map("_id") @db.ObjectId
+    name   String  @unique
+  }
+
+  model Order {
+    id              String   @id @default(auto()) @map("_id") @db.ObjectId
+    product         Product  @relation(fields: [productId], references: [id])
+    shippingAddress Address
+    billingAddress  Address?
+    productId       String   @db.ObjectId
+  }
+
+  type Address {
+    street String @id @default(auto()) @map("_id") @db.ObjectId
+    city   String
+    zip    String
+  }
+  ```
+
+  Then regenerate your Prisma Client using the `npx prisma generate` command. Next, update your application to reflect the new field, and redeploy your app. The records in the database will be updated as the application's users begin to enter their billing address in the new field.
+
+- **"No breaking changes" updates**: this strategy builds on the first one, with further consensus amongst your team that you don't rename or delete fields, only add new fields, and always define the new fields as optional. This policy can be reenforced by adding checks in the CI/CD process to verify that there are no backwards-incompatible changes to the schema.
+
+- **"All-at-once" updates**: this strategy is similar to traditional migrations in relational databases, where all data is updated to reflect the new schema. In the scenario above, you would create a script to add a value for billing addresses to all existing users in your database. You can then make the field a required field in the application because the schema and the data are consistent.
```

Comment:
I think this one could mention a case here this is actually necessary.

Reply:
^ can you clarify this point. I'm not sure what you mean. Are you looking for an example or suggesting that there are times when it's actually necessary to do this kind of update?

Reply:
The first two are possible in the specific case where you add an optional field - old data will still be valid with the new schema. When you do a change that does not adhere to that, and e.g. old data would make Prisma Client explode, you will _need_ to use this approach here. A specific example for a change that makes this required could help people to understand when this is the case.

Reply:
Will revisit improving this after the release.